ADRs/RH/SIG-SRE/ADR-00007-Error-Budget-Policy.md

```diff
@@ -0,0 +1,250 @@
+# ADR-00007 Error Budget Policy
+
+Authors: Craig Robinson
+
+
+## Status
+
+Draft
+
+## Problem Statement
+
+A fundamential tenet of Site Reliability Engineering is to maximise customer satisfaction, not only through the function that a service provides, but also through its stability (aka reliability). This stability is measured through an SLO.
+
+This "SLO miss" is most often described referred to as the "error budget". The error budget can be described as the amount of "error" that a system can withstand without having a negative effect on the service, most notably, the customers "experience" of the service. Mathmatically, it is often described as the inverse of the SLO: that is, as an example, there is an SLO target of 99.9% (3 nines) then the error budget is 0.1%. It can also be referred to as a measure of time (e.g. the error budget is 4 minutes per month).
+
+For an SLO to be effective, it needs to have consequences. That is:
+
+"What happens when we consume all our error budget?"
+
+In many organisations, this question is often answered through the existence of a document, backed by a process, that can be called an "Error Budget Policy".
+
+
+## Goals
+
+The aim of this ADR is to provide a template, with some rationale, for an example Error Budget Policy.
+
+
+## Non-goals
+
+* This ADR does not aim at providing a definitive policy. The intention would be that consumers of this document would be able to take this example, based on the lessons learned from SRE teams already experienced in the running of services, and mold it to their own experience. It is purposefully generalised.
+
+
+## Current Architecture
+
+* An analysis of a number of existing Error Budget Policies was conducted, identifying the commonalities and the differences. The best practices articulated in this document was a synthesis of these sources.
+
+
+## Proposed Architecture
+
+The proposed content is as follows:
+
+***
+
+# Error Budget Policy (draft)
+
+Service Name: &lt;Service/s Name&gt;
+
+# Service
+
+&lt;Briefly describe the service/s under the scope of this document.&gt;
+
+# Definitions
+
+* _SLI_: The Service Level Indicator is the carefully-considered measure for an aspect of the service that is considered necessary to its function (e.g. the login screen should be available).
+* _SLO_: The Service Level Objective is the desired target for the aspect measured by an SLI (e.g. the login screen should be available for 99% of the time).
+* _Error Budget_:  The Error budget is one of the fundamental metrics for service reliability. It is the measure of how much a service can be unreliable (e.g. fail) without impacting customers. Error budgets are always based on SLOs.
```

Comment:
It's not that unreliability within the SLO doesn't impact customers _at all_. It might, but that's deemed acceptable (to the service owner, but also to customers since they've accepted the SLA).

Reply:
Thanks @tombentley . Do you think that needs to be reflected in the text...or it is sufficient (in a general sense) as it stands currently?